### PR TITLE
Add basic jscodeshift demo

### PR DESCRIPTION
**Pull request by malomarrec**

### CODE REVIEW

File: js-codeshift/warn-to-log.batch.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: codeshift-demo
+description: Replace console.warn by console.log
```
Comment (eseliger) [inline js-codeshift/warn-to-log.batch.yml:2]: maybe it would be simpler to include the transformer script in here, so you don't need to rebuild the docker image each time you adjust the script:

```yaml
file: &jscodeshift_file
  /warn-to-log.ts: |
    export default function transformer(file, api) {
      const j = api.jscodeshift;
      const root = j(file.source);
      const consoleLogCalls = root.find(j.CallExpression, {
          callee: {
              object: {
                  name: 'console'
              },
              property: {
                  name: 'warn'
              }
          }
      });
      consoleLogCalls.forEach(p => {
          p.node.callee.property.name = 'log';
      });
      return root.toSource();
    };
name: codeshift-demo
description: Replace console.warn by console.log
# Repositories containing this package
on:
  - repositoriesMatchingQuery: console.warn
steps:
    - run: npx jscodeshift -t /warn-to-log.ts . --extensions js,ts --parser=ts
      files: *jscodeshift_file
      container: malomarrec/codeshift
changesetTemplate:
  title: Replace console.warn by console.log in .js files
  body: Because why not?
  branch: batch-changes/warn-to-log # Push the commit to this branch.
  commit:
    message: js-warn-to-log
  published: false
```

(untested)

File: js-codeshift/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Container image that runs your code
+FROM node:14
+
+RUN npm install -g jscodeshift
+
+# Copies your code file from your action repository to the filesystem path `/` of the container
+COPY warn-to-log.ts /warn-to-log.ts
+
+# Code file to execute when the docker container starts up (`entrypoint.sh`)
+ENTRYPOINT ["jscodeshift", "-t", "./warn-to-log.ts", "./work"]
```
Comment (kkom) [inline js-codeshift/Dockerfile:10]: I've also created a Dockerfile for jscodeshift in https://github.com/kkom/tools-docker/pull/2

So that you can check out the repo and push a versioned image to a docker hub user/organisation of your choice with a command like `make push username=kkom tool=jscodeshift version=0.7.0`

In an enterprise setup you'd probably do it to some kind of internal container registry, but that's convenient to just play with different tools easily

Comment (malomarrec) [reply js-codeshift/Dockerfile:10]: Nice. The docker rebuilding every time you add a transform is quite painful in my demo. I wonder how we can fix this.

File: js-codeshift/warn-to-log.batch.yml (hunk 1 above)
Comment (malomarrec) [reply js-codeshift/warn-to-log.batch.yml:2]: Fixed

File: js-codeshift/warn-to-log.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export default function transformer(file, api) {
```
Comment (eseliger) [inline js-codeshift/warn-to-log.ts:1]: is this file still required then?

File: js-codeshift/warn-to-log.batch.yml (hunk 1 above)
Comment (eseliger) [reply js-codeshift/warn-to-log.batch.yml:2]: oh no 🤦

File: js-codeshift/Dockerfile (hunk 2 above)
Comment (kkom) [reply js-codeshift/Dockerfile]: I'd say maintain an unofficial docker registry of tools, similar to https://www.linuxserver.io/

I'm just going to use https://github.com/kkom/tools-docker/ for my own purposes, but since for now there is no polling to see if a new version is available, I will manually update it as I need it. And of course everyone needs to assess the risk of pulling images from some random `kkom` Docker hub account.

But in all honesty, people can just push the images themselves based on my source code. It won't add any load on Docker Hub, as the layers are deduplicated across accounts based on their hashes.

And anyone can just push a new version whenever they need one. Writing a batch change is an adhoc process anyway.

**Comment by malomarrec** [comment]
Arg. Fixing.

File: js-codeshift/warn-to-log.ts (hunk 3 above)
Comment (malomarrec) [reply js-codeshift/warn-to-log.ts:1]: Fixed in #13
